app/controllers/submit_proposals_controller.rb

```diff
@@ -5,31 +5,48 @@ def new
   end
 
   def create
-    @proposal.update(proposal_params)
-    update_ams_subject_code
-    submission = SubmitProposalService.new(@proposal, params)
-    submission.save_answers
-    session[:is_submission] = @proposal.is_submission = submission.is_final?
+    if @proposal.update(proposal_params)
+      update_ams_subject_code
+      @submission = SubmitProposalService.new(@proposal, params)
+      @submission.save_answers
+      session[:is_submission] = @proposal.is_submission = @submission.is_final?
+
+      response_to_format
```

Comment:
To reduce complexity, the contents of the first conditional (between the if and elsif) could be moved to a separate method.